libraries/AP_Motors/AP_MotorsHeli_Single.cpp

```diff
@@ -490,54 +499,61 @@ void AP_MotorsHeli_Single::output_to_motors()
-                rc_write(AP_MOTORS_MOT_4, calculate_ddfp_output(0.0f));
+                uint16_t ddfp_output = 0;
+                if (ddfp_configured) {
+                    ddfp_output = calculate_ddfp_output(0.0f);
+                }
+                rc_write(AP_MOTORS_MOT_4, ddfp_output);
             }
             break;
         case SpoolState::GROUND_IDLE:
             // sends idle output to motors when armed. rotor could be static or turning (autorotation)
             update_motor_control(ROTOR_CONTROL_IDLE);
             if (_tail_type == AP_MOTORS_HELI_SINGLE_TAILTYPE_DIRECTDRIVE_FIXEDPITCH_CW || _tail_type == AP_MOTORS_HELI_SINGLE_TAILTYPE_DIRECTDRIVE_FIXEDPITCH_CCW){
-                rc_write(AP_MOTORS_MOT_4, calculate_ddfp_output(0.0f));
+                uint16_t ddfp_output = 0;
+                if (ddfp_configured) {
+                    ddfp_output = calculate_ddfp_output(0.0f);
+                }
+                rc_write(AP_MOTORS_MOT_4, ddfp_output);
             }
             break;
         case SpoolState::SPOOLING_UP:
         case SpoolState::THROTTLE_UNLIMITED:
             // set motor output based on thrust requests
             update_motor_control(ROTOR_CONTROL_ACTIVE);
             if (_tail_type == AP_MOTORS_HELI_SINGLE_TAILTYPE_DIRECTDRIVE_FIXEDPITCH_CW || _tail_type == AP_MOTORS_HELI_SINGLE_TAILTYPE_DIRECTDRIVE_FIXEDPITCH_CCW){
-                SRV_Channel *c = SRV_Channels::srv_channel(AP_MOTORS_MOT_4);
-                if (c != nullptr) {
-                    _ddfp_pwm_min = c->get_output_min();
-                    _ddfp_pwm_max = c->get_output_max();
-                    _ddfp_pwm_trim = c->get_trim();
-                }
                 float servo_out = 0.0f;
-                if (is_positive((float) (_ddfp_pwm_max - _ddfp_pwm_min))) {
+                if (ddfp_configured) {
                     float servo4_trim = constrain_float((_ddfp_pwm_trim - 1000) / (_ddfp_pwm_max - _ddfp_pwm_min), 0.0f, 1.0f);
                     if (is_positive(_servo4_out)) {
                         servo_out = (1.0f - servo4_trim) * _servo4_out + servo4_trim;
                     } else {
                         servo_out = servo4_trim * _servo4_out + servo4_trim;
                     }
+                    rc_write(AP_MOTORS_MOT_4, calculate_ddfp_output(thr_lin.thrust_to_actuator(constrain_float(servo_out, 0.0f, 1.0f))));
                 } else {
                     // if servo pwm min and max are bad, convert servo4_out from -1 to 1 to 0 to 1
                     servo_out = 0.5f * (_servo4_out + 1.0f);
+                    // write as an angle and let servo channel class handle the trim min max
+                    rc_write_angle(AP_MOTORS_MOT_4, thr_lin.thrust_to_actuator(constrain_float(servo_out, 0.0f, 1.0f))*4500);
                     // this should never happen
                     INTERNAL_ERROR(AP_InternalError::error_t::flow_of_control);
```

Comment:
We really should not be throwing internal errors at loop rate. Better to add it as a pre-arm.

Reply:
I would just not bother with this whole code path, its a miss configuration we don't have to do anything sensible. Motors arming checks cannot be skipped.